src/gbm/gblinear.cc

```diff
@@ -65,7 +65,11 @@ class GBLinear : public GradientBooster {
     updater_.reset(LinearUpdater::Create(param_.updater, learner_param_));
     updater_->Configure(cfg);
     monitor_.Init("GBLinear");
+    if (param_.updater == "gpu_coord_descent") {
```

Comment:
Can we have a helper function which does those translation and checks instead?

Reply:
@chenqin I'm working toward that.  Full validation for all parameters like whether some parameters are not used.  But right now I need to eliminate the copies of these parameters.